packages/actor-rdf-join-inner-multi-bind/lib/ActorRdfJoinMultiBind.ts

```diff
@@ -147,7 +147,8 @@ export class ActorRdfJoinMultiBind extends ActorRdfJoin<IActorRdfJoinMultiBindTe
         // Send the materialized patterns to the mediator for recursive join evaluation.
         const operation = operations.length === 1 ?
           operations[0] :
-          algebraFactory.createJoin(operations);
+          // Flattening should only take place if none of the input operations have associated metadata
+          algebraFactory.createJoin(operations, operations.every(op => !op.metadata));
```

Comment:
@jitsedesmet Do take note of this when integrating sparqlalgebrajs.
Factory functions like these should probably default to *not* flatten, and only flatten if explicitly requested.
This is to avoid unexpected issues like this.